UIKitDZ/UIKitDZ/SharingViewController.swift

```diff
@@ -0,0 +1,118 @@
+//
+//  ViewController.swift
+//  UIKitDZ
+//
+//  Created by Дмитрий Супрун on 21.09.22.
+//
+
+import UIKit
+/// Sharing different items
+final class SharingViewController: UIViewController {
+    
+    // MARK: - IBOutlet
+    
+    @IBOutlet weak var pickerInputTextField: UITextField!
+    @IBOutlet weak var urlLabel: UILabel!
+    @IBOutlet weak var shareTextField: UITextField!
+    
+    // MARK: - Private properties
+    private let facebook = FacebookShare()
+    private lazy var sharePicker = {
+        let picker = UIPickerView()
+        return picker
+    }()
+    
+    // MARK: - Lifecycle
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        sharePicker.delegate = self
+        sharePicker.dataSource = self
+        pickerInputTextField.inputView = sharePicker
+        let gestureRecognizer = UITapGestureRecognizer(target: self, action: #selector(endEditingAction))
+        view.addGestureRecognizer(gestureRecognizer)
+    }
+    
+    // MARK: - IBAction
+    
+    @IBAction func shareFacebook(_ sender: Any) {
+        shareAction(activityItems: [facebook])
+    }
+    
+    @IBAction func shareURLAction(_ sender: Any) {
+        guard let share = urlLabel.text else { return }
+        shareAction(activityItems: [share])
+    }
+    
+    @IBAction func shareImage(_ sender: Any) {
+        guard let image = UIImage(named: "bagpack") else { return }
+        shareAction(activityItems: [image])
+    }
+    
+    @IBAction func shareTextField(_ sender: Any) {
+        guard let share = shareTextField.text else { return }
+        shareAction(activityItems: [share])
+    }
+    
+    // MARK: - Private Methods
+    
+    private func shareAction(activityItems: [Any], applicationActivities: [UIActivity]? = nil) {
+        let activityViewController = UIActivityViewController(activityItems: activityItems,
+                                                              applicationActivities: applicationActivities)
+        present(activityViewController, animated: true)
+    }
+    
+    @objc func endEditingAction() {
```

Comment:
```suggestion
    @objc func endEditingAction() {
```private func ниже  чем  @objc func и  func